### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by thatmadhacker**

### CODE REVIEW

File: src/ca/team2706/vision/trackerboxreloaded/Main.java [hunk 1]
```diff
@@ -306,34 +411,49 @@ public static void main(String[] args) {
...
                     System.err.println("Error: Failed to get a frame from the camera");
                     continue;
                 }
             } // else use the image from disk that we loaded above
-
-            if (use_GUI)
-                Imgproc.resize( frame, frame, visionParams.sz );
+            if(use_GUI)
+            //Resize the frame
+            Imgproc.resize( frame, frame, visionParams.sz );
```
Comment (ounsworth) [inline src/ca/team2706/vision/trackerboxreloaded/Main.java]: Hmm, I'm not sure if the `if` contains the `Imgproc.resize(..)`, or if it only contains the comment. Better add `{ }`'s to the `if` just to be safe.

File: src/ca/team2706/vision/trackerboxreloaded/Main.java [hunk 2]
```diff
@@ -306,34 +411,49 @@ public static void main(String[] args) {
...
+            	//If use gui then draw the prefered target
+                //Sets the raw image to the frame
+            	rawOutputImg = frame.clone();
+                //Draws the preffered target
+                Pipeline.drawPreferredTarget(rawOutputImg, visionData);
+            } else {
+                //Sets the raw image to the frame
+            	rawOutputImg = frame;
+            }
+            //Sends the data to the vision table
```
Comment (ounsworth) [inline src/ca/team2706/vision/trackerboxreloaded/Main.java]: Hmm, this is actually a code change. You or I will need to go back through the commit history to figure out which is the correct version.

File: src/ca/team2706/vision/trackerboxreloaded/Main.java [hunk 3]
```diff
@@ -306,34 +411,49 @@ public static void main(String[] args) {
...
             if (use_GUI) {
                 try {
                     // May throw a NullPointerException if initializing
                     // the window failed
-                    guiRawImg.updateImage(matToBufferedImage(outputTargetImg));
-                    guiProcessedImg.updateImage(matToBufferedImage(visionData.binMask));
+                	//Update the raw image gui
+                    guiRawImg.updateImage(matToBufferedImage(rawOutputImg));
+                    //Update the processed image gui
+                    guiProcessedImg.updateImage(matToBufferedImage(visionData.outputImg));
```
Comment (ounsworth) [inline src/ca/team2706/vision/trackerboxreloaded/Main.java]: This is also a code change (`outputTargetImg` --> `rawOutputImg`, and `binMask` --> `outputImg`). From memory, I think the `binMask` version is actually correct. This was a change that I made after Ryerson, which I'm guessing wasn't on whichever branch you started your `Documentation` branch from.

File: src/ca/team2706/vision/trackerboxreloaded/Main.java (hunk 2 above)
Comment (ounsworth) [reply src/ca/team2706/vision/trackerboxreloaded/Main.java]: This is might be related to the `binMask` change below, but I don't remember off the top of my head.

File: src/ca/team2706/vision/trackerboxreloaded/Main.java [hunk 4]
```diff
@@ -348,29 +468,36 @@ public static void main(String[] args) {
...
             long elapsedTime = (System.currentTimeMillis() / 1000) - current_time_seconds;
+            //If the elapsed time is more that the seconds between image dumps then dump images asynchronously
             if (elapsedTime >= seconds_between_img_dumps) {
+            	//Sets the current number of seconds
                 current_time_seconds = (System.currentTimeMillis() / 1000);
-
-                Mat finalFrame = frame.clone();
+				//Clones the frame
+				Mat finalFrame = frame.clone();
+				//Starts a new thread to dump images
```
Comment (ounsworth) [inline src/ca/team2706/vision/trackerboxreloaded/Main.java]: Clean up the indentation here?

File: src/ca/team2706/vision/trackerboxreloaded/Main.java [hunk 5]
```diff
@@ -348,29 +468,36 @@ public static void main(String[] args) {
...
+                        	//Dumps the raw image
                             imgDump(matToBufferedImage(finalFrame), "raw");
+                            //Dumps the binMask image
                             imgDump(matToBufferedImage(visionData.binMask), "binMask");
-                            imgDump(matToBufferedImage(outputTargetImg), "output");
+                            //Draw the target to the output image
+                            Pipeline.drawPreferredTarget(finalFrame, visionData);
+                            //Dumps the output image
+                            imgDump(matToBufferedImage(finalFrame), "output");
                         } catch (IOException e) {
```
Comment (ounsworth) [inline src/ca/team2706/vision/trackerboxreloaded/Main.java]: Another code change. Didn't `outputTargetImg` already had the targets drawn on it from [here](https://github.com/FRC2706/TrackerboxReloaded/pull/57/files#diff-051c1874533492a85443dcaa4edd1192L326)?

**Comment by thatmadhacker** [comment]
Addressed all issues, will be tested with the other stuff in a few mins

**Comment by thatmadhacker** [comment]
Tested everything and merged with TimeStamps

**Pull request closed without merging**